opds_import.py

```diff
@@ -260,53 +321,135 @@ def import_from_metadata(
             edition, self.metadata_client, replace=policy
         )
 
+        return edition
+
+
+    def import_pools_works_from_circulationdata(
```

Comment:
I think it would be easier to understand (and your error handling would be cleaner) if you split out the code that creates the Work into a separate method.